src/main/java/ch/uzh/ifi/hase/soprafs22/controller/GameController.java

```diff
@@ -29,35 +30,121 @@
-                                    @PathVariable(value = "id") Long id,
-                                    @RequestBody BlackCardPostDTO blackCardPostDTO) {
-    System.out.println(blackCardPostDTO);
-    userService.checkSpecificAccess(token, id);
+  public void createGame(@RequestHeader(value = "authorization", required = false) String token,
+                                    @PathVariable(value = "userId") Long id,
+                                    @RequestBody CardPostDTO blackCardPostDTO) {
+
+    userService.checkSpecificAccess(token, id); // throws 401, 404
 
     BlackCard userInputCard = DTOMapper.INSTANCE.convertGamePostDTOToEntity(blackCardPostDTO);
 
-    return DTOMapper.INSTANCE.convertEntityToBlackCardGetDTO(userInputCard);
+    //TODO: check if time is up
+
+    // if time is up, create game with game service
+    Game game = gameService.createGame(userInputCard, id);
+    // ad game to user
+    userService.addGame(id, game);
   }
 
+    @GetMapping("users/{userId}/games/blackCards")
+    @ResponseStatus(HttpStatus.CREATED)
+    @ResponseBody
+    public CardGetDTO getBlackCardFromUser(@RequestHeader(value = "authorization", required = false) String token,
+                                      @PathVariable(value = "userId") Long id) {
+
+        userService.checkSpecificAccess(token, id);
+        BlackCard bcRandomUser = userService.getBlackCardFromRandomUser(id);
+
+        return DTOMapper.INSTANCE.convertEntityToCardGetDTO(bcRandomUser);
+    }
+
+    @GetMapping("users/{userId}/games/{gameId}/cards")
+    @ResponseStatus(HttpStatus.CREATED)
+    @ResponseBody
+    public List<CardGetDTO> getCards(@RequestHeader(value = "authorization", required = false) String token,
+                                                @PathVariable(value = "userId") Long id) {
+      // check token
+      userService.checkSpecificAccess(token, id); // throws 401, 404
+      List <WhiteCard> cards = userService.getWhiteCards(id);
+
+      List<CardGetDTO> cardGetDTO= new ArrayList<>();
+      for (WhiteCard card : cards){
+          cardGetDTO.add(DTOMapper.INSTANCE.convertEntityToCardGetDTO(card));
+      }
+
+      return cardGetDTO;
+
+    }
+
+    /**
+     * get a Play with a white card to vote on
+     * and the userId from the user that played that white card
+     */
+    @GetMapping("/users/{userID}/games/{gameId}/vote")
+    @ResponseStatus(HttpStatus.CREATED)
+    @ResponseBody
+    public Play getPlay(@RequestHeader(value = "authorization", required = false) String token,
+                        @PathVariable(value = "gameId") Long gameId, @PathVariable(value = "userId") Long id) {
+
+        userService.checkSpecificAccess(token, id); // throws 401, 404
+
+        // search for random Play with this gameId
+        return gameService.getRandomPlay(gameId);
+    }
+
+    /**
+     * set Like to WhiteCard
+     */
+    @PutMapping("/users/{userID}/games/{gameId}/vote")
```

Comment:
make sure to not have userID in the URI and userId in the pathvariable (not the same)

Reply:
thanks!